bigbluebutton-html5/imports/ui/components/video-provider/component.jsx

```diff
@@ -538,6 +544,20 @@ class VideoProvider extends Component {
     };
   }
 
+  _getOnIceConnectionStateChangeCallback(id) {
+    const { intl } = this.props;
+    const peer = this.webRtcPeers[id];
+
+    return (event) => {
+      const connectionState = peer.peerConnection.iceConnectionState;
+      if (connectionState === 'failed') {
```

Comment:
Seems pretty good overall, but I think we should also check for the `closed` state transition. The idea here is to listen to it and fire an ICE connectivity error if the user isn't in the middle of an unsharing procedure (like we do with audio), because every state can transition to `closed` and that *may* be an error. My suggestion is to add the `closed` listener here too and remove it when an user unshares the webcam, so it'll not trigger the error without a proper reason.